cruisetrack/process/workflow_lines.py

```diff
@@ -4,53 +4,39 @@
 import pandas as pd
 from PyQt5.QtCore import QVariant
 from matplotlib import pyplot as plt
-from qgis.core import QgsField, QgsPointXY, QgsFeatureRequest, QgsPoint
+from qgis.core import QgsField, QgsPointXY, QgsPoint, QgsVectorLayer
 
+from cruisetrack.process import plot_track
 
-def process_lines(layer_provider, laye_r, is_individual_trackline, is_accessory, is_nonebt,
-                  is_normal_profile, flip_we, only_process_2nds, is_littorina, flip_ns):
-    field_name = "X_start"  # if there are the right fields still missing, make them
-    field_index = laye_r.fields().indexFromName(field_name)
-    if field_index == -1:
-        layer_provider.addAttributes([QgsField("X_start", QVariant.Double),
-                                      QgsField("X_stop", QVariant.Double), QgsField("Y_start", QVariant.Double),
-                                      QgsField("Y_stop", QVariant.Double), QgsField("length", QVariant.Double),
-                                      QgsField("X_mean", QVariant.Double)])
-        laye_r.updateFields()
 
-    # put data from shapefile into attribute table
-    coun_t = -1
+def line_features_to_df(laye_r) -> pd.DataFrame:
+    row_list = []
     for fea_t in laye_r.getFeatures():
         geom = fea_t.geometry()
-        coun_t = coun_t + 1
+        for part in geom.get():  # https://gis.stackexchange.com/a/304805
+            first_vertex = part[0]
+            last_vertex = part[-1]
+        row_list.append({
+            'X_start': first_vertex.x(),
+            'X_stop': last_vertex.x(),
+            'Y_start': first_vertex.y(),
+            'Y_stop': first_vertex.y(),
+            'length': geom.length(),
+            'X_mean': geom.centroid().asPoint().x(),
+            'Y_mean': geom.centroid().asPoint().y(),
+        })
+    df = pd.DataFrame(row_list)
+    return df
 
-    """ if just one line, with multiple points, 
-    but as normal regular profile lines wanted (additinoally accessory tasks)"""
-    if coun_t == 0:
+
+def layer_to_dataframe(laye_r: QgsVectorLayer, single_line: bool, is_individual_trackline: bool):
```

Comment:
This one is reading the layer into a dataframe. That way there is no layer needed later in the actual processing